### ISSUE TITLE
[FEATURE REQ] ATOM API Allow configuring "rule" while creating subscription

### ISSUE THREAD

**Issue by HarshaNalluru** [Azure/azure-sdk-for-java#17384]
.NET SDK provides CreateSubscription overload which allows configuring the rule.
```C#
public virtual Task<Response<SubscriptionProperties>> CreateSubscriptionAsync(CreateSubscriptionOptions options, CancellationToken cancellationToken = default);
public virtual Task<Response<SubscriptionProperties>> CreateSubscriptionAsync(CreateSubscriptionOptions options, CreateRuleOptions rule, CancellationToken cancellationToken = default);
```

This issue is to track adding support for the same in Java V7 SDK.
This feature is important because..
While creating a new subscription to an active topic(messages are actively being sent), there is no way users can stop filling up the subscription before the rule filter is added(since the default rule filter is TRUE).

Original issue: https://github.com/Azure/azure-sdk-for-js/issues/12345

@YijunXieMS @hemanttanwar @conniey @ramya-rao-a

**Comment by ramya-rao-a** [comment]
Moving the milestone to Backlog as this can be added in the next semester as well. Please feel free to add details on how to approach this task and add the "help wanted" label

**Comment by ki1729** [comment]
Feature out for review: https://github.com/Azure/azure-sdk-for-java/pull/31804

**Comment by ki1729** [comment]
Thisis now merged and will be released as part of SDK version 7.13.0
